src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -20,5 +24,62 @@ internal static void VerifyValidPath(string path, string argName)
                 throw new ArgumentException(SR.Argument_InvalidPathChars, argName);
             }
         }
+
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            SearchOption searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+            bool success = true;
+
+            try
+            {
+                directoryEnumeration = Directory.EnumerateDirectories(sourcePath, "*", searchOption);
+            }
+            catch (IOException)
+            {
+                // Return false if Enumeration is not possible
+                return false;
+            }
```

Comment:
In my opinion it would probably be better if you use the [other overload](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratedirectories?view=net-6.0#system-io-directory-enumeratedirectories(system-string-system-string-system-io-enumerationoptions)) for `EnumerateDirectories` which uses `EnumerationOptions` and then it should contain at least `IgnoreInaccessible`. Otherwise this function fails without copying what can be copied if just a sub directory enumeration fails. This will probably be the first that people are going to report as an issue. Maybe @bartonjs agrees with me?

Reply:
@Symbai IgnoreInaccessible seems to be the default...
https://github.com/dotnet/runtime/blob/59e643b9932122a12e71ceb360a9ae3bb9420dec/src/libraries/System.Private.CoreLib/src/System/IO/EnumerationOptions.cs#L32-L37

Reply:
Ah I see... The default uses EnumerationOptions.Compatible. I'll update this